### PR TITLE
DEVDOCS-4630: [update] Carts, update coupons datatype

### PR DESCRIPTION
**Pull request by bc-traciporter**
# [DEVDOCS-4630]

## What changed?
I added the option for coupons to be a number or an array for Physical and Digital items.

## Anything else?
Related PRs, salient notes, etc

[DEVDOCS-4630]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4630?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bc-mackxu — DISMISSED** [review]
Thank you Traci. LGTM

**Comment by slsriehl** [comment]
At first glance -- based on some of the changes I'm seeing that stoplight introduced, I'd like a full search through for apostrophes that are single-quote unicode characters. @bc-traciporter

File: reference/carts.v3.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-openapi: '3.0.3'
```
Comment (slsriehl) [inline reference/carts.v3.yml:1]: this should retain single quotes. the datatype of the `openapi` value is `string`.

Comment (bc-traciporter) [reply reference/carts.v3.yml:1]: @slsriehl I added the single quotes for `openapi` but I don't see where else I need to add them. Also, other files do not have single quotes for `openapi` which needs to be added.  I can create a ticket for these.

**Review by slsriehl — DISMISSED** [review]
changes requested per our conversation today.

File: reference/carts.v3.yml [hunk 2]
```diff
@@ -1711,13 +1709,11 @@ components:
                     type: number
                     description: The product option value identifier in number format.
                     example: 128
-                title: Product Option
           required:
             - variant_id
             - product_id
             - quantity
-        - type: object
```
Comment (kzhang-dsg) [inline reference/carts.v3.yml:1719]: `type: object` is required and should not be removed.

File: reference/carts.v3.yml [hunk 3]
```diff
@@ -2093,22 +2081,8 @@ components:
             - variant_id
             - product_id
             - quantity
-        - properties:
-            is_require_shipping:
```
[the remark this answers is not in the document]
Comment (bc-traciporter) [reply reference/carts.v3.yml:2097]: @kzhang-dsg If you are looking now, you will not see the changes until this PR is merged into main.

File: reference/carts.v3.yml (hunk 2 above)
Comment (bc-traciporter) [reply reference/carts.v3.yml:1719]: @kzhang-dsg I see `type: object` on line 1697

Comment (kzhang-dsg) [reply reference/carts.v3.yml:1719]: For some reason I dont see type: object on line 1697. I check ed both the file in this commit and the file in the main branch.

See the file in this commit: https://github.com/bigcommerce/api-specs/blob/e0d7b063ea77955a71b590ad09357cbff75adf8a/reference/carts.v3.yml
image

**Pull request merged**